redash/query_runner/impala_ds.py

```diff
@@ -0,0 +1,137 @@
+                    "type": "string"
+                },
+                "host": {
+                    "type": "string"
+                },
+                "port": {
+                    "type": "number"
+                },
+                "host": {
+                    "type": "string"
+                },
+                "use_ldap": {
+                    "type": "boolean"
+                },
+                "ldap_user": {
+                    "type": "string"
+                },
+                "ldap_user": {
+                    "type": "string"
+                },
+                "ldap_password": {
+                    "type": "string"
+                },
+                "timeout": {
+                    "type": "number"
+                }
+            },
+            "required": ["host"]
+        }
+
+    @classmethod
+    def type(cls):
+        return "impala"
+
+    def __init__(self, configuration_json):
+        super(Impala, self).__init__(configuration_json)
+
+    def get_schema(self):
+        #todo: implement :-)
+        pass
+
+    def run_query(self, query):
+        #connection = connect(host=self.configuration['host'],
+        #                     port=(self.configuration['port'] if 'port' in self.configuration else None),
+        #                     protocol=(self.configuration['protocol'] if 'protocol' in self.configuration else None),
+        #                     timeout=(self.configuration['timeout'] if 'timeout' in self.configuration else None),
+        #                     database=(self.configuration['database'] if 'database' in self.configuration else None),
+        #                     use_ldap=(self.configuration['use_ldap'] if 'use_ldap' in self.configuration else None),
+        #                     ldap_user=(self.configuration['ldap_user'] if 'ldap_user' in self.configuration else None),
+        #                     ldap_password=(self.configuration['ldap_password'] if 'ldap_password' in self.configuration else None),)
+        connection = connect(**self.configuration)
+
+        cursor = connection.cursor()
+
+        try:
+            cursor.execute(query)
+
+            column_names = []
+            columns = []
+
+            for column in cursor.description:
+                column_name = column[COLUMN_NAME]
+                column_names.append(column_name)
+
+                columns.append({
+                    'name': column_name,
+                    'friendly_name': column_name,
+                    'type': types_map.get(column[COLUMN_TYPE], None)
+                })
+
+            rows = [dict(zip(column_names, row)) for row in cursor]
+
+            data = {'columns': columns, 'rows': rows}
+            json_data = json.dumps(data, cls=JSONEncoder)
+            error = None
+            cursor.close()
+        except KeyboardInterrupt:
+            connection.cancel()
+            error = "Query cancelled by user."
+            json_data = None
+        except Exception as e:
```

Comment:
What kind of exception Impala will throw in case of syntax error? Might want to return it as error instead of rethrowing.